AUTOMATAFORMAT.md

```diff
@@ -0,0 +1,111 @@
+* A file contains **sections** starting with a line of the form `@<SECTION-TYPE>`, containing an automaton or, in theory, anything. We will start with one section per file, containing an automaton, but it is obviously extensible to more sections and more kinds of things.
+* In automata sections, non-empty lines are **key-value lines** of the form `%<KEY> [VALUE]`, or **transition lines**. Several key-value lines with the same key mean that the key is mapped to the set of values, an occurence without a value marks the `KEY` as defined. 
+* Besides white spaces and the end of line, the following are **reserved characters**: `&`,`|`,`!`,`@`,`(`,`)`,`%`,`"`,`/`,`\` `[`,`]`,`a`,`q`,`n`,`f`,`{`,`}`.
+  * `&`,`|`,`!`,`(`,`)` occurring as tokens in transition lines are logical operators.
+  * `[`,`]`,`-` within intervals of symbols of the form `[bla-bli]`.
+  * `{`,`}` are enclosing attributes, if we use them this way. Remains to be decided. (?)
+  * `@` opens the line with a section name and is used in transducer alphabet tokens of the form `x@a1`, `y@[10-55]`,...
+  * `%` opens a key-value line, possibly also key-value appendix of a transition. Remains to be decided. (?)
+  * `"` strings containing white spaces and reserved characters can be written in between `"`. The reserved characters lose their special meaning. The characters `"` and `\` inside a string must be escaped, i.e. `\"` and `\\`.
+  * `/` to concatenate lines.
+  * `\` for escaping characters.
+  * `a`,`q`,`n`,`f` are token type specifiers (alphabet, state, node, formula). If you don't like these, propose other ones.
+* The parser should recognise the reserved symbols as standalone tookens only if they appear in the special context where their special meaning applies, otehrwise they should be treated as normal symbols (is this reasonable ?).
+
+We categorise automata by **transition type**, that is, the structure of their transitions, and by **alphabet type**, i.e., how alphabet symbols on transitions are represented. It is expected that different parameters would be parsed into different data structures. These parameters should determine the name of the automata section.
+
+## Automata by transition type: 
+We use general form of AFA, by Pavol V., and simple NFA.
+
+### AFA 
+* Each state has a transition, a booelan formula over symbol literals, states, and nodes.
+* Several lines starting with the same state mean the disjunciton of the formulae. No line for the state means false. 
+* Since in the general formulae, the type of a token cannot be determined by its position, we require one of the three schemes for distinguishing token types.
+  * By **enumeration**.
+  * By **type-markers**. The literal is preceeded by a single character that specifies the type, we use `q` for states, `a` for symbols, and `n` for shared nodes. The marker is *not* a part of the thing's name! (Or should it be? It would perhaps be simpler that way?)
```

Comment:
I'm for including the marker into name (for simplicity and transparency)

Reply:
Agree with Pavol, it would be better to include marker into name.

Reply:
It gets little problematic with formulae, like fx=y things. There the f cannot be a part of the actual formula. But formulas are not the most important thing right now. Can you elaboreate on why it is simpler and more transparent?

Reply:
However, including into the name seem to create a problem with convertability of formats. Since the names of typed states must start with `q`, one cannot convert names that do not start with q into it. The same for other things. One would have to remember that the q is an additional stuff ... It sounds as a reason to not include it to me.

Reply:
Another thing, with `"`s, would one write things like `q"blabla bla"` or `"qblabla bla"`? I would rather use the first option, but that looks like that "q" is not a part of the state.

Reply:
Overall, excluding the type marker might be the more logical option, maybe worth it even though some things will look more ugly.

Reply:
Ok, type markers are not part of name. This is definitive decision.